sim/hunter/apl_values.go

```diff
@@ -0,0 +1,98 @@
+package hunter
+
+import (
+	"fmt"
+
+	"github.com/wowsims/sod/sim/core"
+	"github.com/wowsims/sod/sim/core/proto"
+)
+
+func (hunter *Hunter) NewAPLValue(rot *core.APLRotation, config *proto.APLValue) core.APLValue {
+	switch config.Value.(type) {
+	case *proto.APLValue_HunterPetIsActive:
+		return hunter.newValueHunterPetIsActive(rot)
+	case *proto.APLValue_HunterCurrentPetFocus:
+		return hunter.newValueHunterCurrentPetFocus(rot, config.GetHunterCurrentPetFocus())
+	case *proto.APLValue_HunterCurrentPetFocusPercent:
+		return hunter.newValueHunterCurrentPetFocusPercent(rot, config.GetHunterCurrentPetFocusPercent())
+	default:
+		return nil
+	}
+}
+
+type APLValueHunterPetIsActive struct {
+	core.DefaultAPLValueImpl
+	hunter *Hunter
+}
+
+func (hunter *Hunter) newValueHunterPetIsActive(_ *core.APLRotation) core.APLValue {
+	return &APLValueHunterPetIsActive{
+		hunter: hunter,
+	}
+}
+
+func (value *APLValueHunterPetIsActive) Type() proto.APLValueType {
+	return proto.APLValueType_ValueTypeBool
+}
+func (value *APLValueHunterPetIsActive) GetBool(sim *core.Simulation) bool {
+	return value.hunter.pet != nil
```

Comment:
We could go a step further here and say `value.hunter.pet != nil && value.hunter.pet.IsActive()` since pets can be deactivated